origin-js/src/models/offer.js

```diff
@@ -3,24 +3,25 @@
 //
 export class Offer {
   /**
-   * An Offer is constructed based on its on-chain and off-chain data.
-   * @param {string} offerId - Unique offer ID.
+   * Offer object model.
+   *
+   * @param {string} id - Offer ID.
    * @param {string} listingId - Unique listing ID.
-   * @param {Object} chainOffer - Offer data from the blockchain.
-   * @param {Object} ipfsOffer - Offer data from IPFS.
+   * @param {string} status - Satus of the offer: 'created', 'accepted', 'disputed', 'finalized', 'sellerReviewed'
+   * @param {int} createdAt - Time in seconds since epoch
+   * @param {string} buyer - address of the buyer
+   * @param {Array{Object}} events - list of events ( like OfferCreated event)
+   * @param {string} refund - Amount to refund buyer upon finalization
    */
-  constructor(offerId, listingId, chainOffer, ipfsOffer) {
-    this.id = offerId
-    this.listingId = listingId
-    this.status = chainOffer.status // 'created', 'accepted', 'disputed', 'finalized', 'sellerReviewed'
-    this.createdAt = chainOffer.createdAt // Time in seconds since epoch.
-    this.buyer = chainOffer.buyer
-    this.events = chainOffer.events
-    this.refund = chainOffer.refund
-    this.totalPrice = chainOffer.totalPrice
-
-    // See src/schemas/offer.json for fields stored in IPFS offer data.
-    Object.assign(this, ipfsOffer)
+  constructor(id, listingId, status, createdAt, buyer, events, refund, totalPrice) {
```

Comment:
Hmmm... That's a pretty long list of arguments and with JS not being good at doing any check on arg type and number of args, it could get pretty easy for caller to mess up... Also take a look at the Listing models use case, it has even more arguments...
Passing a dictionary with fields set to the various arg would probably be less error prone in my opinion.

And I don't think it would necessarily be bad to have a couple of "helper" methods for building an Offer model. One based on (chain + ipfs data), the other based on discovery data.Just to make it easier for the caller. Like Offer.init(chainData, ipfsData) and Offer.initDisco(discoData). And those would call the constructor. Just an idea...

Thanks for working on this and trying to make that core Origin code as clean and simple as possible !

Reply:
Yeah that is a good point, no named arguments and type checks can quickly result in errors. I like your idea with helper methods that construct the object and a single constructor better. Changing to that, thanks!